torch/nn/functional.py

```diff
@@ -1178,6 +1178,47 @@ def local_response_norm(input, size, alpha=1e-4, beta=0.75, k=1):
     return input / div
 
 
+def instance_norm_wrapper(input, weight, bias, saved_running_mean, saved_running_var,
+                          training=False, momentum=0.1, eps=1e-5, affine=False):
+    """This wrapper is for ONNX exporting.
+    In PyTorch, InstanceNorm is implemented using BatchNorm. However it is not a good idea to export
+    InstanceNorm as a combination of Reshapes and BatchNorm to ONNX, since Reshapes may not be cheap
+    in other frameworks, plus we don't want to lose high-level information during exporting. So here
+    we apply the same trick/hack as we did for RNN. This trick requires that the symbolic function
+    and the real implementation must have exactly the same signatures.
+    """
+    import torch
+    func = instance_norm
+    if torch._C._jit_is_tracing(input):
```

Comment:
in general symblic_override does invoke inside of it a check on _jit_is_tracing. I think the only problem is that it checks all args and Adam was complaining that flattening is not perf-cheap.

Maybe modify symbolic_override to look only at the first argument (if it's a tensor), or have symbolic_override_based_on_first? Then this whole function can be a simple decorator on `instance_norm` function and would look much nicer

Reply:
I like symbolic_override_based_on_first more, @ezyang, what do you think? :-)